x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/remove_layer_button.tsx

```diff
@@ -39,27 +96,168 @@ export function RemoveLayerButton({
+
+  const removeLayer = () => {
+    // If we don't blur the remove / clear button, it remains focused
+    // which is a strange UX in this case. e.target.blur doesn't work
+    // due to who knows what, but probably event re-writing. Additionally,
+    // activeElement does not have blur so, we need to do some casting + safeguards.
+    const el = document.activeElement as unknown as { blur: () => void };
+
+    if (el?.blur) {
+      el.blur();
+    }
+
+    onRemoveLayer();
+  };
+
   return (
-    <EuiButtonIcon
-      size="xs"
-      iconType={isOnlyLayer ? 'eraser' : 'trash'}
-      color="danger"
-      data-test-subj="lnsLayerRemove"
-      aria-label={ariaLabel}
-      title={ariaLabel}
-      onClick={() => {
-        // If we don't blur the remove / clear button, it remains focused
-        // which is a strange UX in this case. e.target.blur doesn't work
-        // due to who knows what, but probably event re-writing. Additionally,
-        // activeElement does not have blur so, we need to do some casting + safeguards.
-        const el = document.activeElement as unknown as { blur: () => void };
-
-        if (el?.blur) {
-          el.blur();
-        }
-
-        onRemoveLayer();
-      }}
-    />
+    <>
+      <EuiButtonIcon
+        size="xs"
+        iconType={isOnlyLayer ? 'eraser' : 'trash'}
+        color="danger"
+        data-test-subj="lnsLayerRemove"
+        aria-label={ariaLabel}
+        title={ariaLabel}
+        onClick={() => {
+          if (lensLocalStorage?.skipDeleteModal) {
+            return removeLayer();
+          }
+          return showModal();
+        }}
+      />
+      {isModalVisible ? (
+        <RemoveConfirmModal
+          modalTitle={modalTitle}
+          isDeletable={!!activeVisualization.removeLayer && !isOnlyLayer}
+          modalDesc={modalDesc}
+          closeModal={closeModal}
+          skipDeleteModal={lensLocalStorage?.skipDeleteModal}
+          onChangeShouldShowModal={onChangeShouldShowModal}
+          removeLayer={removeLayer}
+        />
+      ) : null}
+    </>
   );
 }
+
+const RemoveConfirmModal = ({
+  modalTitle,
+  modalDesc,
+  closeModal,
+  skipDeleteModal,
+  onChangeShouldShowModal,
+  isDeletable,
+  removeLayer,
+}: {
+  modalTitle: string;
+  modalDesc: string;
+  closeModal: () => void;
+  skipDeleteModal?: boolean;
+  isDeletable?: boolean;
+  onChangeShouldShowModal: () => void;
+  removeLayer: () => void;
+}) => {
+  return (
+    <EuiModal data-test-subj="lnsLayerRemoveModal" onClose={closeModal}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>{modalTitle}</EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <p>
+          {modalDesc}
+          {i18n.translate('xpack.lens.layer.confirmModal.cannotUndo', {
+            defaultMessage: `You can't undo this action.`,
+          })}
+        </p>
+      </EuiModalBody>
+
+      <EuiModalFooter>
+        <EuiFlexGroup justifyContent="spaceBetween">
```

Comment:
Can we vertically align this group's flex items to be centered?

```suggestion
        <EuiFlexGroup alignItems="center" justifyContent="spaceBetween">
```